apis/v1alpha2/gateway_types.go

```diff
@@ -764,12 +792,20 @@ const (
 	// the Listener could not be attached to the Gateway because the
 	// requested address is not supported.
 	ListenerReasonUnsupportedAddress ListenerConditionReason = "UnsupportedAddress"
+
+	// This reason is used with the "Detached" condition when the condition is
+	// false.
+	ListenerReasonAttached ListenerConditionReason = "Attached"
```

Comment:
Yeah that was the intent, the reason would be "Attached" any time Detached was false. I agree it's rather confusing. My initial proposal was that these negative conditions simply shouldn't be set when they're false. The [API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) do state that unset == unknown though, which is not really what we'd be trying to communicate if we left these unset.

Reply:
Ok that's what I thought that they always need to be there, thanks for clarifying. 

I wonder if we could document this somehow, so that it's clear all the different conditions should be present on each type at all times.

Reply:
Or do they get set by default when the object is created? That would be super cool, then you wouldn't need to deal with the initial state from the controller.